### PR TITLE
V4 base branch

### PR DESCRIPTION
**Pull request by ealsur**
# Pull Request Template

## Description

V4 preview initial base branch

### CODE REVIEW

File: Microsoft.Azure.Cosmos.sln [hunk 1]
```diff
@@ -9,6 +9,12 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Azure.Cosmos.Test
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Azure.Cosmos.EmulatorTests", "Microsoft.Azure.Cosmos\tests\Microsoft.Azure.Cosmos.EmulatorTests\Microsoft.Azure.Cosmos.EmulatorTests.csproj", "{F16D0FDA-36A1-420C-BA6B-BB30FA62B28D}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Azure.Cosmos", "Microsoft.Azure.Cosmos\azuredata\Azure.Cosmos.csproj", "{5F4D16C2-4FB0-4775-846B-92CFFE9F2A00}"
```
Comment (j82w) [inline Microsoft.Azure.Cosmos.sln]: Is there any reason we aren't putting this in a separate repo?

File: Microsoft.Azure.Cosmos/azuredata/Azure.Cosmos.csproj [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <ShippingScope>External</ShippingScope>
+    <SigningType>Product</SigningType>
+    <DebugType>portable</DebugType>
+    <IncludeSymbols>false</IncludeSymbols>
+    <IncludeSource>false</IncludeSource>
+    <RootNamespace>Azure.Cosmos</RootNamespace>
+    <NoWarn>NU5125</NoWarn>
+    <LangVersion>8.0</LangVersion>
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/azuredata/Azure.Cosmos.csproj]: Are we suppose to target such a new LangVersion? Will this not hurt adoption by customer using older VS versions.

Comment (ealsur) [reply Microsoft.Azure.Cosmos/azuredata/Azure.Cosmos.csproj]: Cannot target lower versions, `IAsyncEnumerable` is only available in >8

File: Microsoft.Azure.Cosmos.sln (hunk 1 above)
Comment (ealsur) [reply Microsoft.Azure.Cosmos.sln]: V4 is linking most of the files from V3, so any change in V3 is easily ported

File: Microsoft.Azure.Cosmos.sln [hunk 3]
```diff
@@ -9,6 +9,12 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Azure.Cosmos.Test
...
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Azure.Cosmos.EmulatorTests", "Microsoft.Azure.Cosmos\tests\Azure.Cosmos\Azure.Cosmos.EmulatorTests\Azure.Cosmos.EmulatorTests.csproj", "{0C673EC9-9951-4A25-B0BB-AE16E256530B}"
```
[the remark this answers is not in the document]
Comment (ealsur) [reply Microsoft.Azure.Cosmos.sln]: Working on the YML files, mainly the release ones first (you can see them below on the PR running)

**Pull request merged**